samples: tfm: Run tfm tests when tfm or mcuboot modules change

d3zd3z opened the pull request:
Now that CI supports running tests, via tag, on changes that modify a module, add the tags necessary to the TF-M tests so that these tests are run in any PR where tfm or mcuboot are modified.  This should reduces the cases where an mcuboot change breaks tfm, and this is not caught.

d3zd3z committed:
samples: tfm: Update tag to run tests on module change

A recent change #61718, adds support to the twister build to check for
manifest changes, and run tests using tags based on the modules that have
changed.  The tfm tests have an existing tag `tfm` to select these tests,
but the above patch uses a tag named after the module, trusted-firmware-m.
To make this work, add `trusted-firmware-m` in addition to the existing tag
`tfm` so that these tests will run whenever the tfm module is updated.

Signed-off-by: David Brown <david.brown@linaro.org>

d3zd3z committed:
samples: tfm: Also test tfm when mcuboot changes

In addition to running the tfm tests when the trusted-firmware-m module
changes, most of the tests also use MCUboot, which comes from our module.
Add the mcuboot tag to each of these tests, except for the single test that
explicitly builds without using MCUboot.

Signed-off-by: David Brown <david.brown@linaro.org>

nashif commented on samples/tfm_integration/psa_crypto/sample.yaml:
```diff
@@ -5,7 +5,7 @@ sample:
 tests:
   sample.psa_crypto:
-    tags: introduction tfm crypto csr
+    tags: introduction tfm trusted-firmware-m crypto csr mcuboot
```
please, convert this to a yaml list.

nashif commented on samples/tfm_integration/psa_protected_storage/sample.yaml:
```diff
@@ -28,4 +28,4 @@ common:
 tests:
   sample.tfm.protected_storage:
-    tags: tfm
+    tags: tfm trusted-firmware-m mcuboot
```
ditto

nashif commented on samples/tfm_integration/tfm_ipc/sample.yaml:
```diff
@@ -7,6 +7,8 @@ tests:
       - introduction
       - tfm
+      - trusted-firmware-m
+      - mcuboot
```
yes, like here :)

nashif commented on samples/tfm_integration/tfm_psa_test/sample.yaml:
```diff
@@ -1,5 +1,5 @@
 common:
-  tags: tfm
+  tags: tfm trusted-firmware-m mcuboot
```
list please

nashif commented on samples/tfm_integration/tfm_regression_test/sample.yaml:
```diff
@@ -1,5 +1,5 @@
 common:
-  tags: tfm
+  tags: tfm trusted-firmware-m mcuboot
```
list please

nashif commented on samples/tfm_integration/tfm_secure_partition/sample.yaml:
```diff
@@ -1,5 +1,5 @@
 common:
-  tags: tfm
+  tags: tfm trusted-firmware-m
```
convert to yaml list

nashif commented on samples/tfm_integration/tfm_secure_partition/sample.yaml:
```diff
@@ -22,4 +22,4 @@ sample:
 tests:
   sample.tfm.secure_partition:
-    tags: tfm
+    tags: tfm trusted-firmware-m mcuboot
```
convert to list please.

doesn't `tfm` stand for `trusted-firmware-m`?

d3zd3z replied:
tfm does stand for trusted-firmware-m, but I don't know if people might have external tests or things that already use this tag. I need to add the trusted-firmware-m tag so that it matches the module name.

joerchan commented:
When should we use 'tfm' and when 'trusted-firmware-m'.
Are these different tags, or can we combine?

microbuilder commented:
> When should we use 'tfm' and when 'trusted-firmware-m'.
> 
> Are these different tags, or can we combine?

I think at some point early on there were two tags, but today we only have one so these should likely be updated to reflect that.

d3zd3z commented:
My concern with removing `tfm` is that we don't know if anyone externally is using that tag to run their tests. I don't think it hurts anything to have both tags.

microbuilder commented:
> My concern with removing `tfm` is that we don't know if anyone externally is using that tag to run their tests. I don't think it hurts anything to have both tags.

I doubt any tests are being run on these outside Zephyr's CI? I could be wrong, but it would surprise me.

joerchan commented:
> > My concern with removing `tfm` is that we don't know if anyone externally is using that tag to run their tests. I don't think it hurts anything to have both tags.
> 
> I doubt any tests are being run on these outside Zephyr's CI? I could be wrong, but it would surprise me.

I think at Nordic we may do this, but should be easy to update to. I think it is better to do this than have the two tags, which can cause problems and confusion when you end up not using the right tag.

d3zd3z committed:
samples: tfm: Remove old 'tfm' tag

To avoid confusion having both a 'tfm' and a 'trusted-firmware-m' tag,
remove the old 'tfm' tag entirely.  Any external test frameworks that are
using the 'tfm' tag will need to be changed to account for this.

Signed-off-by: David Brown <david.brown@linaro.org>
